src/ArborX_LinearBVH.hpp

```diff
@@ -235,6 +248,45 @@ BoundingVolumeHierarchy<MemorySpace, Enable>::BoundingVolumeHierarchy(
   Kokkos::Profiling::popRegion();
 }
 
+template <typename MemorySpace, typename Enable>
+template <typename ExecutionSpace, typename Predicates, typename Callback>
+void BoundingVolumeHierarchy<MemorySpace, Enable>::query(
+    ExecutionSpace const &space, Predicates const &predicates,
+    Callback const &callback, Experimental::TraversalPolicy const &policy) const
+{
+  Details::check_valid_access_traits(PredicatesTag{}, predicates);
+  using Access = AccessTraits<Predicates, PredicatesTag>;
+  static_assert(KokkosExt::is_accessible_from<typename Access::memory_space,
+                                              ExecutionSpace>::value,
+                "Predicates must be accessible from the execution space");
+
+  Details::check_valid_callback(callback, predicates);
+
+  Kokkos::Profiling::pushRegion("ArborX::BVH::query");
+
+  auto const &bvh = *this;
```

Comment:
Same here, I really dislike your aliasing `*this`

Reply:
:shrug: